### ISSUE TITLE
Completion for string property doesn't include replacementSpan if string is unclosed 

### ISSUE THREAD

**Issue by stefan-lacatus** [microsoft/TypeScript#54092]
monaco-editor version: 0.10.1
Browser: All
OS: All
Steps or JS usage snippet reproducing the issue:
Completions of fields with dot in their name is somewhat broken.
How to reproduce:

1. You have the following object: `let x = { "foo.bar": "test" };`
2. Typing in `x["fo]` and pressing ctrl+space, and then selecting `foo.bar` generates the expected result
3. Typing in `x["foo.b` and pressing ctrl+space, and selecting `foo.bar` generates `x["foo.foo.bar`.  

Basically if there is a dot in the field name the issue appears. It only happens in monaco editor. VScode does not have the same bug.

I would gladly fix this if you can point me in the right direction (monaco-typescript most likely, in the completions provider, but i don't really see what the issue is in the code).

**Comment by mjbvz** [comment]
Repo code:

```ts
const a = { "foo.bar": 1}
a["foo.b 
```

The returned `completionInfo` doesn't have a `replacementSpan` in this case:

```
 "entries": [
        {
            "name": "foo.bar",
            "kind": "property",
            "kindModifiers": "",
            "sortText": "11"
        }
    ]
```

However if you have a trailing `"`, the a `replacementSpan` is returned:

```
    "entries": [
        {
            "name": "foo.bar",
            "kind": "property",
            "kindModifiers": "",
            "sortText": "11",
            "replacementSpan": {
                "start": {
                    "line": 2,
                    "offset": 4
                },
                "end": {
                    "line": 2,
                    "offset": 9
                }
            }
        }
    ]
```
